src/pages/gd/Claim/index.tsx

```diff
@@ -29,19 +31,26 @@ const Claim = memo(() => {
     // 3. If neither is true, there is a claim ready for user or its a new user and FV will be triggered instead
     useEffect(() => {
         if (claimAmount?.isZero()) {
+            setRefreshRate('never')
             setClaimed(true)
         }
         // after just having claimed and switching chains,
         // the state of transaction might still be cached causing the ui to update incorrectly
         // why we force a reset of the tx state after it completes
         else if (state.status === 'Success') {
+            setRefreshRate('everyBlock')
             resetState()
             setClaimed(true)
         } else {
             setClaimed(false)
         }
     }, [claimAmount, state, send, chainId])
 
+    // upon switching chain we want temporarily to poll everyBlock up untill we have the latest data
+    useEffect(() => {
+        setRefreshRate(claimAmount ? 'never' : 'everyBlock')
```

Comment:
you are missing claimAmount as dependency. but actually this should not use claimAmount at all.
if chainId changed you always refresh.
your other effects will turn off the refresh once you get the refreshed claimAmount and state for the new chain

Reply:
we are missing the last use case where the timer reaches 0 and a new claim cycle started, we need to refresh also then.
the simplest approach would be to set the default refresh to 12 instead of 'never' so if user is waiting in the screen it will at least refresh it once in a while.
don't forget that we use the hook refreshOrNever which turns off refresh if tab is not active